lib/buffer.js

```diff
@@ -1021,557 +1002,764 @@ Buffer.prototype.slice = function slice(start, end) {
   return new FastBuffer(this.buffer, this.byteOffset + start, newLength);
 };
 
+function checkOffsetType(offset) {
+  if (typeof offset !== 'number') {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                               'offset',
+                               'number',
+                               offset);
+  }
+}
+
+function boundsError(offset, ext, length) {
+  if (Math.floor(offset) !== offset || offset < 0) {
+    throw new errors.RangeError('ERR_OUT_OF_RANGE',
+                                'offset',
+                                'Math.floor(offset) === offset && offset >= 0');
+  }
 
-function checkOffset(offset, ext, length) {
-  if (offset + ext > length)
+  if (offset + ext > length) {
     throw new errors.RangeError('ERR_INDEX_OUT_OF_RANGE');
+  }
 }
 
-function checkByteLength(byteLength) {
-  if (byteLength < 1 || byteLength > 6) {
-    throw new errors.RangeError('ERR_OUT_OF_RANGE',
-                                'byteLength',
-                                '>= 1 and <= 6');
+function byteLengthError(byteLength) {
+  if (typeof byteLength !== 'number') {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                               'byteLength',
+                               'number',
+                               byteLength);
   }
+  throw new errors.RangeError('ERR_OUT_OF_RANGE',
+                              'byteLength',
+                              '>= 1 and <= 6');
 }
 
+function checkInt(value, offset, min, max) {
+  checkOffsetType(offset);
+  if (value > max || value < min) {
+    throw new errors.RangeError('ERR_INVALID_OPT_VALUE', 'value', value,
+                                `Range: value <= ${max} && value >= ${min}`);
+  }
+}
 
-Buffer.prototype.readUIntLE =
-  function readUIntLE(offset, byteLength, noAssert) {
-    offset = offset >>> 0;
-    byteLength = byteLength >>> 0;
-    if (!noAssert) {
-      checkByteLength(byteLength);
-      checkOffset(offset, byteLength, this.length);
-    }
+Buffer.prototype.readUIntLE = function readUIntLE(offset, byteLength) {
+  if (byteLength === 6)
```

Comment:
This should be better as a switch statement, no?

Reply:
I personally prefer this style as it is a bit less to read. The performance is the same. If you have a strong feeling about it, I'll change it, otherwise I would rather keep it.